python/Ganga/Core/InternalServices/ShutdownManager.py

```diff
@@ -74,6 +74,8 @@ def _ganga_run_exitfuncs():
         from Ganga.Core.MonitoringComponent.Local_GangaMC_Service import getStackTrace
         getStackTrace()
         monitoring_component.disableMonitoring()
+        monitoring_component.stop()
+        monitoring_component.join()
```

Comment:
The sooner the better in shutdown imo to stop extra threads starting up and getting in the way